### DISCUSSION TITLE
Proposal: Decentralized Voting Mechanism for the Batch

### DISCUSSION THREAD

**Discussion by fedeloterstein** [BuidlGuidl/batch9.buidlguidl.com#24] [Ideas]
After reading the messages in the Telegram group, I realized we need a decentralized voting mechanism to help us make decisions and solve the issues we’re facing in this Batch.

I’m attaching evidence of the problem:

<img width="551" alt="Captura de pantalla 2024-09-13 a la(s) 00 03 11" src="https://github.com/user-attachments/assets/506cb74d-ae61-4aca-b63f-9e66a2a82aa2">

<img width="551" alt="Captura de pantalla 2024-09-13 a la(s) 00 02 47" src="https://github.com/user-attachments/assets/78f8c9a6-8202-4e59-beeb-8f8818558bd8">

We all agree that a decision needs to be made, so I came up with this idea:

<img width="875" alt="Captura de pantalla 2024-09-13 a la(s) 00 00 43" src="https://github.com/user-attachments/assets/53605e81-7489-4e26-838f-3149967b3100">

Proposal: Voting Smart Contract
I’m suggesting we create a smart contract that integrates with our BatchRegistry.sol contract to allow registered users to vote on proposals. Key features would include:

Who can vote: Only users who have done the check-in on BatchRegistry.sol can vote.
One vote per proposal: Each user can vote only once per proposal.
Proposals to vote on: The contract should allow registering different PRs for voting.
Deadline: There should be a deadline for voting.
Idea for the Interface
Additionally, I suggest a simple widget on the homepage to make voting easy and accessible for everyone.
<img width="479" alt="Captura de pantalla 2024-09-13 a la(s) 00 12 00" src="https://github.com/user-attachments/assets/e103ccee-4a06-43ee-b8b0-cbab83f32a52">

Opening the discussion to see if you’re interested in the idea. I’d love to hear your feedback! This could even be the beginning of a DAO structure for our decision-making process.

**Comment by d4rm5** [comment]
This idea is very cool! The only thing is that I'm afraid it's too much for the dynamics of the batch. I wouldn't want decisions that are normally made quickly to be delayed by the implementation of the voting system.

**Comment by arjanjohan** [comment]
I have to agree, the idea is nice and interesting to build. However, simple decisions like this don't benefit from such a process.

In this case, we can just merge the first PR for the logo and close the others. In case it's needs to be updated at a later point, we can open a new issue/PR.
